include/eve/detail/function/simd/common/patterns.hpp

```diff
@@ -0,0 +1,63 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/traits/cardinal.hpp>
+#include <eve/traits/as_wide.hpp>
+#include <ostream>
+
+namespace eve::detail
+{
+  //================================================================================================
+  // Helpers for detecting patterns
+  //================================================================================================
+
+  // Detects <N,...,N> as a broadcast
+  template<int I0, int... I> inline constexpr bool is_broadcast = ((I0 !=-1) && ... && (I0==I));
+
+  // Detects <-1,...,-1> as zeroing
+  template<int... I> inline constexpr bool is_zero = ((I==-1) && ...);
```

Comment:
Why zero and not just identity?
Is -1 zero? Or "I don't care"?

This -1 should be a meaningful enum

We seem to be inconsistent again about what we support where, no? is everything supports -1s?

Reply:
Fixed